codespan/src/filemap.rs

```diff
@@ -15,6 +15,30 @@ pub enum FileName {
     Virtual(Cow<'static, str>),
 }
 
+impl From<PathBuf> for FileName {
+    fn from(name: PathBuf) -> FileName {
+        FileName::real(name)
+    }
+}
+
+impl<'a> From<&'a Path> for FileName {
+    fn from(name: &Path) -> FileName {
+        FileName::real(name)
+    }
+}
+
+impl From<String> for FileName {
+    fn from(name: String) -> FileName {
+        FileName::virtual_(name)
+    }
+}
+
+impl From<&'static str> for FileName {
+    fn from(name: &'static str) -> FileName {
+        FileName::virtual_(name)
+    }
+}
```

Comment:
Awesome stuff. I wasn't sure about adding these, but I think it's fine.

Reply:
Yeah, I don't think they should be that surprising and it was a bit tedious to need to import `FileName` frequently when I really just want to give a name to files added with https://github.com/brendanzab/codespan/blob/ccff649894f5ba3ce8c1c4c823a9ca2798d90ec8/codespan/src/codemap.rs#L31